File: core/clientLaya/game/src/commonGame/net/base/CenterRequest.ts
namespace Shine
{
	export class CenterRequest extends BaseRequest
	{
		constructor()
        {
            super();
			//关了检测
			this.setOpenCheck(false);
        }

		protected doWriteToStream(stream:BytesWriteStream)
		{
			//直接协议号和内容
			stream.natureWriteUnsignedShort(this._dataID);
			this.doWriteBytesSimple(stream);
		}
	
		/** 发送 */
		public send():void
		{
			GameC.player.sendCenter(this);
		}
	}
}

File: core/client/game/src/commonGame/net/base/CenterRequest.cs
using System;
using ShineEngine;

/// <summary>
/// 中心服消息
/// </summary>
public class CenterRequest:BaseRequest
{
	public CenterRequest()
	{
		//关了检测
		setNeedFullWrite(ShineSetting.clientMessageUseFull);
	}

	protected override void doWriteToStream(BytesWriteStream stream)
	{
		//直接协议号和内容
		stream.natureWriteUnsignedShort(_dataID);
		doWriteBytesSimple(stream);
	}

	/** 发送 */
	public void send()
	{
		GameC.player.sendCenter(this);
	}
}